lib/puppet_x/puppetlabs/powershell/compatible_powershell_version.rb

```diff
@@ -0,0 +1,50 @@
+require File.join(File.dirname(__FILE__), 'powershell_version')
+
+module PuppetX
+  module PuppetLabs
+    module PowerShell
+      class CompatiblePowerShellVersion
+        def self.compatible_version?
+          value = false
+
+          # PowerShell version should be checked - if we are using
+          # PowerShell v3+, we are good to go b/c .NET 4+
+          # https://msdn.microsoft.com/en-us/powershell/scripting/setup/windows-powershell-system-requirements
+          # Look at Microsoft .NET Framwork Requirements section.
+          powershell_version = PuppetX::PuppetLabs::PowerShell::PowerShellVersion.version
+          return false if powershell_version.nil?
+
+          if Gem::Version.new(powershell_version) >= Gem::Version.new(3)
+            return true
+          end
+
+          # If we are using PowerShell v2, we need to see what the latest
+          # version of .NET is that we have
+          # https://msdn.microsoft.com/en-us/library/hh925568.aspx
+          if Puppet::Util::Platform.windows?
+            require 'win32/registry'
+
+            begin
+              # At this point in the check, PowerShell is using .NET Framework
+              # 2.x family, so we only need to verify v3.5 key exists.
+              # If we were verifying all compatible types we would look for
+              # any of the following:
+              # - v3.5 key
+              # - then v4.0
+              # - then v4
+              hive = Win32::Registry::HKEY_LOCAL_MACHINE
+              # redirection doesn't actually matter here - disable it anyway
+              hive.open('SOFTWARE\Microsoft\NET Framework Setup\NDP\v3.5', Win32::Registry::KEY_READ | 0x100) do |reg|
+                value = true
+              end
```

Comment:
~~So we don't support .Net 3.0 either?~~  .Net 3.0 is no longer supported.  3.5 SP1 is the supported version in the 3.x family.